From bbcd4d526da87da6733a76c64ee84b7f112209cf Mon Sep 17 00:00:00 2001
From: mqi <mqi@codeaurora.org>
Date: Sun, 4 Jun 2017 20:03:01 +0800
Subject: [PATCH 17/19] frameworks: Fix null pointer Issue

Add a null pointer check to avoid crash while customizing quick settings
in systemui.

CRs-Fixed: 1094610
Change-Id: I03eaaaeeb8bbf7fbc91ca09eb16dd1ab559d2d6f
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
---
 core/java/android/widget/ArrayAdapter.java | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/core/java/android/widget/ArrayAdapter.java b/core/java/android/widget/ArrayAdapter.java
index f554f895a9..172d702c35 100644
--- a/core/java/android/widget/ArrayAdapter.java
+++ b/core/java/android/widget/ArrayAdapter.java
@@ -448,7 +448,9 @@ public class ArrayAdapter<T> extends BaseAdapter implements Filterable, ThemedSp
         }
 
         final T item = getItem(position);
-        if (item instanceof CharSequence) {
+        if (item == null) {
+            text.setText("");
+        } else if (item instanceof CharSequence) {
             text.setText((CharSequence) item);
         } else {
             text.setText(item.toString());
-- 
2.34.1

